src/frequenz/sdk/timeseries/_quantities.py

```diff
@@ -275,6 +279,38 @@ def __eq__(self, other: object) -> bool:
         return self._base_value == other._base_value
 
 
+_T = TypeVar("_T")
+
+
+def disable_default_constructor(cls: type[_T]) -> type[_T]:
+    """Disable the default constructor for a class.
+
+    Check that the class does not already have a default constructor, and add one that
+    raises a TypeError if called.
+
+    Args:
+        cls: The class to disable the default constructor for.
+
+    Returns:
+        The class with the default constructor disabled.
+
+    Raises:
+        TypeError: If the class already has a default constructor.
+    """
+    if "__init__" in cls.__dict__:
+        raise TypeError(f"No default constructor allowed for {cls.__name__}")
+
+    def __init__(self: Any, *args: Any, **kwargs: Any) -> None:
+        raise TypeError(
+            f"Use of default constructor NOT allowed for {cls.__name__}. "
+            f"Use one of the `{cls.__name__}.from_*()` methods instead."
+        )
+
+    cls.__init__ = __init__  # type: ignore[method-assign]
+    return cls
```

Comment:
I'm a bit scared now about patching classes via a decorator now, after the headaches this brought with the `@actor` decorator. I guess for this case where you are only adding a `__init__` that probably won't confuse `mypy` it should be OK, but I fear in the future someone might think it is a good idea to add some other common method here and things go south again.

Other approaches that might be more `mypy` friendly are just defining a base type `_NoDefaultConstructible` and inherit from it (but that won't catch if the error if someone adds an `__init__` to it, although it would be weird to inherit from `_NoDefaultConstructible` and add an `__init__`. But another alternative would be metaclasses, I'm not sure how `mypy` feels about those, but I think is the most widely method used to manipulate classes, so my guess is that probably better than decorators.

For example: https://stackoverflow.com/questions/8212053/private-constructor-in-python/64682734#64682734

This also adds the `_create()` method that could actually be useful to abstract sub-classes from dealing with `__new__()` directly.

Reply:
I've replaced it with a meta class.  Didn't add a private constructor,  because it hast to be typed with `Any`,  whereas with `__new__`, we get proper typing.